dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -354,10 +365,46 @@ def initialize_storage(self, truncate_tables: Iterable[str] = None) -> None:
             logger.info(f"Will truncate tables {truncate_tables}")
             self.truncate_tables(list(truncate_tables))
 
+        # if the init file already exists, we return
+        if self.fs_client.exists(self.pathlib.join(self.dataset_path, INIT_FILE_NAME)):
```

Comment:
you are creating the path to the init file multiple times, you can do it once, this reduces duplication.